### ISSUE TITLE
Static pods be evicted when diskPressure. Commit c05d506 doesn't seem to work.

### ISSUE THREAD

**Issue by KeepTheBeats** [kubernetes/kubernetes#83317]
**What happened**:

My k8s version is 1.15.
I want my static pods which are not in "kube-system" won't be evicted when diskPressure.
I found that after PR: https://github.com/kubernetes/kubernetes/pull/80491/files ,  static pods should not be evicted when diskPressure.
I won't update to 1.16 temporarily.
So I cherry-picked this commit: c05d5060193221dd5ebcf031d2b2708af5d00c5f to my kubelet.

But when I did some tests, I found that this commit doesn't work when diskPressure.
I think the reason is like info in this comment: https://github.com/kubernetes/kubernetes/issues/73572#issuecomment-470444663

detail:

When diskPressure , the pod sent to `IsCriticalPod` is a mirrorPod.(`pkg/kubelet/eviction/eviction_manager.go#evictPod`).

When a mirrorPod list-watched by kubelet, the `annotation[kubernetes.io/config.source]` is set to be the source that kubelet list-watch the pod from, which is `"api"`. (`pkg/kubelet/config/config.go#updatePodsFunc`)

Because the `annotation[kubernetes.io/config.source]` of  mirrorPod is `"api"`, the func `"pkg/kubelet/types/pod_update.go#IsStaticPod"` returns `false`. 
So static pods will be evicted when diskPressure.

I thought how to fix it.
The func `"pkg/kubelet/types/pod_update.go#IsCriticalPod"` deals with staticPod and mirrorPod in different scene. 
So I changed my kubelet to add the verifying of mirrorPod after the verifying of staticPod.
just like this(My original code is k8s1.15 cherry-pick c05d5060193221dd5ebcf031d2b2708af5d00c5f, so it looks strange.):
```go
func IsCriticalPod(pod *v1.Pod) bool {
	if IsStaticPod(pod) {
		return true
	}
	if IsMirrorPod(pod) {
		return true
	}
	if utilfeature.DefaultFeatureGate.Enabled(features.PodPriority) {
		if pod.Spec.Priority != nil && IsCriticalPodBasedOnPriority(*pod.Spec.Priority) {
			return true
		}
	}
	if utilfeature.DefaultFeatureGate.Enabled(features.ExperimentalCriticalPodAnnotation) {
		if IsCritical(pod.Namespace, pod.Annotations) {
			return true
		}
	}
	return false
}
```

After I changed my code to this, static pod didn't be evicted when diskPressure.
So I think this is the reason. And I think this is a Bug.

Maybe I will set `pod.Spec.Priority` to my static pods to prevent eviction temporarily, and I also hope this bug will be fixed in a good way.

**What you expected to happen**:
Static pods won't be evicted when diskPressure.

**How to reproduce it (as minimally and precisely as possible)**:
1. Run a static pod on a node. The pod is not set `pod.Spec.Priority` or `pod.Spec.PriorityClassName`. 
2. Let the node under diskPressure (using `fallocate -l 104G file1`).

Then the static pod will be evicted.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): My kubelet is k8s1.15  cherry-pick c05d5060193221dd5ebcf031d2b2708af5d00c5f.
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by tedyu** [comment]
Raised PR #83325

**Comment by dchen1107** [comment]
Thanks for reporting the issue. One question: Is that static pod killed / evicted? Or just corresponding mirror pod was marked as Evicted without a real kill / eviction? If it is the latter, Kubelet should reconsolidate the status later. But I agreed that we should fix the issue since there is a race window when both the admission control and eviction managemer would think there might be more resource space.

**Comment by KeepTheBeats** [comment]
> Thanks for reporting the issue. One question: Is that static pod killed / evicted? Or just corresponding mirror pod was marked as Evicted without a real kill / eviction? If it is the latter, Kubelet should reconsolidate the status later. But I agreed that we should fix the issue since there is a race window when both the admission control and eviction managemer would think there might be more resource space.

I did a test just now. 
(My kubelet now is 1.15 version, with no c05d5060193221dd5ebcf031d2b2708af5d00c5f, but I think it makes no difference, because this commit only changed the result of func `pkg/kubelet/eviction/eviction_manager.go#IsCriticalPod`, and whether c05d5060193221dd5ebcf031d2b2708af5d00c5f exists or not, we will get `IsCriticalPod` returns `false` in our scene.)

In the test, after the static pod be evicted, 
1. The corresponding mirror pod was marked as Evicted; 
2. I can't find the containers or sandbox of the static pod use `docker ps -a`. 
So the static pod was truly evicted.

And we can see that in func  `pkg/kubelet/eviction/eviction_manager.go#evictPod`. It sends the `mirror pod` to the func `IsCriticalPod` and sends the `static pod` to the func `killPodFunc`. So from the code, the static pod was truly killed too.

**Comment by andy-wolf168** [comment]
> Thanks for reporting the issue. One question: Is that static pod killed / evicted? Or just corresponding mirror pod was marked as Evicted without a real kill / eviction? If it is the latter, Kubelet should reconsolidate the status later. But I agreed that we should fix the issue since there is a race window when both the admission control and eviction managemer would think there might be more resource space.

Hi,@dchen1107 @tedyu could you fix the issue on 1.15?
